common/models/user.js

```diff
@@ -27,6 +28,36 @@ function getAboutProfile({
   };
 }
 
+function getStatsProfile({
+  username,
+  challengeMap = {}
+}) {
+  return {
+    username,
+    stats: challengeMapToStats(challengeMap)
+  };
+}
+
+// @return array of
+// { date: $commit_date,
+//   url: $repo_url, (optional)
+//   hash: $commit_hash
+// }
+function challengeMapToStats(obj) {
+  var commit = [];
+  Object.keys(obj).map((k) => {
+    if (obj[k].hasOwnProperty('completedDate') &&
+        obj[k].hasOwnProperty('files')) {
+      commit.push({
+        date: obj[k].completedDate,
+        hash: crypto.createHash('sha1')
```

Comment:
@raisedadead I can remove `sha1` completely from this api and move it to the git-stats module to get the shasum. If you prefer, I will update my commit. Should I do this?